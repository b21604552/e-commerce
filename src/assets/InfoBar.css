.info-wrapper {
    display: inline-flex;
    justify-content: space-between;
    align-items: center;
    width: 1280px;
    height: 52px;
    margin-left: 80px;
    margin-right: 80px;
    margin-top: 16px;
}

.category {
    font-style: normal;
    font-weight: bold;
    font-size: 28px;
    line-height: 28px;
    color: #484848;
}

.search-keyword-wrapper {
    display: inline-flex;
}

.search-keyword-label {
    font-style: normal;
    font-weight: normal;
    font-size: 15px;
    line-height: 15px;
    display: flex;
    align-items: center;
    color: #B0B0B0;
    margin-right: 5px;
}

.search-keyword {
    font-style: normal;
    font-weight: normal;
    font-size: 15px;
    line-height: 15px;
}

.filter-wrapper {
    width: 120px;
    height: 48px;
}

.filter-button {
    width: 120px;
    height: 48px;
    border: 1px solid #B0B0B0;
    box-sizing: border-box;
    border-radius: 4px;
    font-style: normal;
    font-weight: normal;
    font-size: 17px;
    line-height: 24px;
    display: flex;
    align-items: center;
    text-align: center;
    justify-content: center;
    color: #B0B0B0;
}

select {
    background-color: white;
    border: 1px solid #B4B4BB;
    border-radius: 4px;
    display: flex;
    justify-content: center;
    align-items: center;
    color: #B0B0B0;
    font-style: normal;
    font-weight: normal;
    font-size: 17px;
    line-height: 24px;
    text-align: center;
    width: 120px;
    padding-right: 20px;
    padding-left: 5px;
    height: 48px;
    margin: 0;
    white-space: nowrap;
    overflow: hidden;
    cursor: pointer;
    text-overflow: ellipsis;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    -webkit-appearance: none;
    -moz-appearance: none;
}

select.custom-select {
    background-image: linear-gradient(45deg, transparent 50%, #BDBDBD 50%), linear-gradient(135deg, #BDBDBD 50%, transparent 50%);
    background-position: calc(100% - 15px) calc(1em + 5px), calc(100% - 10px) calc(1em + 5px);
    background-size: 5px 5px, 5px 5px, 1px 1.5em;
    background-repeat: no-repeat;
}

select.custom-select:focus {
    background-image: linear-gradient(45deg, transparent 50%, #BDBDBD 50%), linear-gradient(135deg, #BDBDBD 50%, transparent 50%);
    background-position: calc(100% - 15px) calc(1em + 5px), calc(100% - 10px) calc(1em + 5px);
    background-size: 5px 5px, 5px 5px, 1px 1.5em;
    background-repeat: no-repeat;
    outline: 0;
}