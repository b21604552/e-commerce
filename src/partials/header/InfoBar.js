import React from 'react'
import { reduxStore } from '../../services/redux/ReduxIndex'
import { setOrderValue } from '../../services/redux/actions/GeneralAction'

export default function InfoBar () {
  const { headerTitle } = reduxStore.getState().GeneralReducer
  const { searchKeyword } = reduxStore.getState().GeneralReducer
  const { orderValue } = reduxStore.getState().GeneralReducer

  function setOrder (e) {
    setOrderValue(e)
  }

  return (
    <div className={'info-wrapper'}>
      <div className="category-wrapper">
        <div className={'category'}>
          {headerTitle}
        </div>
        <div className={'search-keyword-wrapper'}>
          {
            searchKeyword != null && (
              <>
                <div className={'search-keyword-label'}>
                  Aranan Kelime:
                </div>
                <div className={'search-keyword'}>
                  {searchKeyword}
                </div>
              </>
            )
          }
        </div>
      </div>
      <div className="filter-wrapper">
        <select className="custom-select" value={orderValue} onChange={(e) => setOrder(e.target.value)}>
          <option value="" disabled={true} hidden>Sıralama</option>
          <option value="0">En Düşük Fiyat</option>
          <option value="1">En Yüksek Fiyat</option>
          <option value="2">En Yeniler (A>Z)</option>
          <option value="3">En Yeniler (Z>A)</option>
        </select>
      </div>
    </div>
  )
}