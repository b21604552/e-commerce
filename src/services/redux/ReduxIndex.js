import { createStore } from 'redux'
import reducerCombined from './reducerCombined'

export const reduxStore = createStore(reducerCombined)
