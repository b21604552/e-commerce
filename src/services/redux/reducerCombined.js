import { combineReducers } from 'redux'
import GeneralReducer from './reducers/GeneralReducer'

const reducerCombined = combineReducers({
  GeneralReducer: GeneralReducer
})

export default reducerCombined
