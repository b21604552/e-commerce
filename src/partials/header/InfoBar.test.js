import React from 'react'
import ReactDOM from 'react-dom'
import InfoBar from './InfoBar'
import renderer from 'react-test-renderer'
import { reduxStore } from '../../services/redux/ReduxIndex'
import { Provider } from 'react-redux'

describe('InfoBar Test', () => {
  let page = (
    <Provider store={reduxStore}>
      <InfoBar/>
    </Provider>
  )

  it('Renders InfoBar without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(page, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders InfoBar correctly', () => {
    const div = renderer.create(page).toJSON()
    expect(div).toMatchSnapshot()
  })
})