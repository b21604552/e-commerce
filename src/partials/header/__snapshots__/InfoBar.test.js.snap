// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBar Test Renders InfoBar correctly 1`] = `
<div
  className="info-wrapper"
>
  <div
    className="category-wrapper"
  >
    <div
      className="category"
    >
      Hoşgeldiniz
    </div>
    <div
      className="search-keyword-wrapper"
    />
  </div>
  <div
    className="filter-wrapper"
  >
    <select
      className="custom-select"
      onChange={[Function]}
      value=""
    >
      <option
        disabled={true}
        hidden={true}
        value=""
      >
        Sıralama
      </option>
      <option
        value="0"
      >
        En Düşük Fiyat
      </option>
      <option
        value="1"
      >
        En Yüksek Fiyat
      </option>
      <option
        value="2"
      >
        En Yeniler (A&gt;Z)
      </option>
      <option
        value="3"
      >
        En Yeniler (Z&gt;A)
      </option>
    </select>
  </div>
</div>
`;
